shared/src/validation-schemas/user/user-sign-up.validation-schema.ts

```diff
@@ -1,18 +1,35 @@
 import * as Joi from 'joi';
 import { getNameOf } from '~/helpers/helpers';
 import { UserSignUpRequestDto } from '~/common/types/types';
-import { UserValidationMessage } from '~/common/enums/enums';
+import {
+  UserValidationMessage,
+  UserValidationRule,
+} from '~/common/enums/enums';
+
+const NAME_PATTERN = /^[ A-Za-z0-9-'`]*$/;
```

Comment:
```suggestion
```
it should be a part of `UserValidationRule` enum. you can create constant enum (instead of using `enum` keyword)